### PR TITLE
Prevent crashes caused by characters in SIDs which are invalid in paths

### PR DESCRIPTION
**Pull request by EuniverseCat**

### CODE REVIEW

File: Celeste.Mod.mm/Patches/MountainModel.cs [hunk 1]
```diff
@@ -74,7 +74,13 @@ class patch_MountainModel : MountainModel {
         public extern void orig_Update();
         public new void Update() {
             orig_Update();
-            string path = Path.Combine("Maps", SaveData.Instance?.LastArea.GetSID() ?? "").Replace('\\', '/');
+            string path;
+            try {
+                path = Path.Combine("Maps", SaveData.Instance?.LastArea.GetSID() ?? "").Replace('\\', '/');
+            }
+            catch (ArgumentException) {
```
Comment (0x0ade) [inline Celeste.Mod.mm/Patches/MountainModel.cs]:
```suggestion
            } catch (ArgumentException) {
```

File: Celeste.Mod.mm/Patches/MountainModel.cs [hunk 2]
```diff
@@ -98,7 +104,13 @@ class patch_MountainModel : MountainModel {
                 vanillaMoonParticles = (Engine.Scene as Overworld)?.Entities.OfType<MoonParticle3D>().First();
             }
 
-            string path = Path.Combine("Maps", SaveData.Instance?.LastArea.GetSID() ?? "").Replace('\\', '/');
+            string path;
+            try {
+                path = Path.Combine("Maps", SaveData.Instance?.LastArea.GetSID() ?? "").Replace('\\', '/');
+            }
+            catch (ArgumentException) {
```
Comment (0x0ade) [inline Celeste.Mod.mm/Patches/MountainModel.cs]:
```suggestion
            } catch (ArgumentException) {
```

File: Celeste.Mod.mm/Patches/MountainModel.cs [hunk 3]
```diff
@@ -112,7 +124,13 @@ class patch_MountainModel : MountainModel {
                         newMountain = meta.Mountain;
                     }
                 }
-                string oldPath = Path.Combine("Maps", PreviousSID ?? "").Replace('\\', '/');
+                string oldPath;
+                try {
+                    path = Path.Combine("Maps", PreviousSID ?? "").Replace('\\', '/');
+                }
+                catch (ArgumentException) {
```
Comment (0x0ade) [inline Celeste.Mod.mm/Patches/MountainModel.cs]:
```suggestion
                } catch (ArgumentException) {
```

File: Celeste.Mod.mm/Patches/MountainModel.cs [hunk 4]
```diff
@@ -112,7 +124,13 @@ class patch_MountainModel : MountainModel {
...
+                    path = "Maps";
+                }
```
Comment (0x0ade) [inline Celeste.Mod.mm/Patches/MountainModel.cs]: This change seems to cause a build failure. Did you mean to assign to `oldPath` instead of `path` here?

**Review by 0x0ade — CHANGES_REQUESTED** [review]

File: Celeste.Mod.mm/Patches/MountainModel.cs (hunk 4 above)
Comment (EuniverseCat) [reply Celeste.Mod.mm/Patches/MountainModel.cs]: you ever make a typo in your first pr to a repo that more than 2 people work on

**Review by 0x0ade — APPROVED** [review]

**Pull request merged**